datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -126,19 +61,13 @@ pub fn coerce_types(
         | Operator::LtEq
         | Operator::IsDistinctFrom
         | Operator::IsNotDistinctFrom => comparison_coercion(lhs_type, rhs_type),
-        // interval - timestamp is an erroneous case, cannot coerce a type
         Operator::Plus | Operator::Minus
-            if is_datetime(lhs_type)
-                || is_datetime(rhs_type)
-                || is_interval(lhs_type)
-                || is_interval(rhs_type) =>
+            if is_interval(lhs_type) && is_interval(rhs_type) =>
         {
-            if is_interval(lhs_type) && is_datetime(rhs_type) && *op == Operator::Minus {
-                return Err(DataFusionError::Plan(
-                    "interval can't subtract timestamp/date".to_string(),
-                ));
-            }
```

Comment:
Original code in here exist two motivation
- get result type
- get common type

I separate it.